docs/data-sources/directory_object.md

```diff
@@ -0,0 +1,35 @@
+---
+subcategory: "Base"
+---
+
+# Data Source: azuread_directory_object
+
+Retrieves the OData type for a generic directory object having the provided object ID.
+
+## API Permissions
+
+The following API permissions are required in order to use this data source.
+
+When authenticated with a service principal, this data source requires either `Group.Read.All` or `Directory.Read.All`
+
+When authenticated with a user principal, this data source does not require any additional roles.
+
+## Example Usage
+
+*Look up and output type of object by ID*
+```terraform
+data "azuread_directory_object" "example" {
+  object_id = "object-id"
```

Comment:
We like to use syntactically appropriate but obviously-fake values in example configs :)
```suggestion
  object_id = "00000000-0000-0000-0000-000000000000"
```